bazel/external/quiche.BUILD

```diff
@@ -4207,13 +4262,11 @@ envoy_cc_library(
 envoy_cc_library(
     name = "quiche_common_platform_default_quiche_platform_impl_lib",
     srcs = [
-        "quiche/common/platform/default/quiche_platform_impl/quic_mutex_impl.cc",
         "quiche/common/platform/default/quiche_platform_impl/quiche_time_utils_impl.cc",
     ],
     hdrs = [
-        "quiche/common/platform/default/quiche_platform_impl/quic_mutex_impl.h",
```

Comment:
I don't understand why there was a build target with platform defaults for mutex and testvalue here, while :quic_platform_base_impl_lib had Envoy's quic_testvalue_impl.h and :quic_platform_impl_lib had Envoy's quic_mutex_impl.h.  I would assume that any QUICHE consumer would only need to build one or the other (default in QUICHE or local impl), not both.  It feels like a pre-existing inconsistency (maybe leftover from when default implementation was added) and I support fixing it.